### ISSUE TITLE
Some (newer) Haxe language features are not supported

### ISSUE THREAD

**Issue by johhnry** [vantreeseba/tree-sitter-haxe#18]
Hi (again),

Here are the recent features that are not well handled by the parser:

- [`implements`](https://haxe.org/manual/types-interfaces.html) keyword with interfaces:

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/3e036a54-db1b-4eeb-896a-aff97f119c9b)

- the [`final`](https://haxe.org/manual/types-class-inheritance.html#since-haxe-4.0.0) access modifier on classes:

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/e0853e65-8b7d-45db-a84a-5258d3e34d70)

- Generic parameter with the `Array` type (for example) in a method (same for `typedef Ints = Array<Int>;`):

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/d4548665-5e43-4b3f-98e1-c46d2ffc02b2)

- Usage of `this` inside a class method:

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/b377da41-f9dd-456b-a76f-28b663348594)

- `ERROR` in if/else statement:

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/68641ce5-7d6c-44c0-b8d9-985ceb7bcf1e)

And also (giving error when inspecting the tree-sitter tree):

- Passing [anonymous structures](https://haxe.org/manual/types-anonymous-structure.html) to method arguments:

```haxe
interface ITest {
  function accept(json:{a:Int, b:Float}):Bool;
}
```

- [Array comprehension](https://haxe.org/manual/lf-array-comprehension.html) with range expression:

```haxe
var a = [for (i in 0...5) i];
```

- the `using` keyword for [static extensions](https://haxe.org/manual/lf-static-extension.html):

```haxe
using ArrayExtensions;
```

- Using property read/write access (the example is from the [documentation](https://haxe.org/manual/class-field-property.html)):

```haxe
class Main {
	public var x(default, null):Int;

	static public function main() {}
}
```

- [Abstract classes](https://haxe.org/manual/types-abstract-class.html):

```haxe
abstract class Vehicle {}
```

![image](https://github.com/vantreeseba/tree-sitter-haxe/assets/35373357/fe43af92-49e8-442e-bc13-41680fdd6df6)

Thanks!

**Comment by vantreeseba** [comment]
Hey thanks for the clean feedback, I'll try to get to some of this soon.

**Comment by vantreeseba** [comment]
I closed this with PR, version is on NPM @ 0.4.0.

Let me know if you see any of these again, there should be tests covering all cases, if not, I would love if you could just add the tests (in the tests/corpus/*.txt whereever it seems to make sense) and make a PR even if it's failing just to help me ensure all cases are covered.

Anyhow, this should fix all these, and thanks again for the through bug report.

**Comment by johhnry** [comment]
@vantreeseba thank you so much it will improve the DX a lot!
